### DISCUSSION TITLE
Mini guide - How to change jailmaker config and apply any changes with least issues

### DISCUSSION THREAD

**Discussion by mooglestiltzkin** [Jip-Hop/jailmaker#143] [General]
Maybe it's obvious to some, but thought i'd mention it anyway in case others were wondering.

after i had deployed the jailmaker config, i realized i needed to make further changes. So what now?

I initially thought i had to remove the docker jail first, then reinstall the docker config again.

But if you do that, you have to reinstall all your docker containers. E.g. cd to the docker compose directory, do a docker compose up -d
 then using dockge you deploy the dockers. This is what i tested.

However there is a better and easier way to do this.

Just go to truenas shell, do a `jlmkr edit docker` (the jailname i made is docker fyi)

then make your edit, write, then exit. Then `jlmkr restart docker`

if no problems it should startup. if there are issues, just repeat the edit, undo the changes, and restart.

Make sure you save a backup of any working config.

Worse case scenario you have to delete then remake the jail again. Not a total disaster but it's a pain to recover x-x; so don't waste your time.

**Comment by Lockszmith-GH** [comment]
That is a good tip.

Another option to consider, a bit more advanced - if the changes you want to make will affect the installation of the host (for example apt-get calls, etc.)

You can prepare a dataset that is specifically for the `/var/lib/docker` path, and maybe even one for the actual docker-compose files and data.

This way removing a recreating will not touch any docker/docker compose deployment as long as those datasets are bound in your configuration template.

**Comment by mooglestiltzkin** [comment]
credits to stux, he was the one that first let me know. and ty lock for the add on
